### PR TITLE
feat(api): add external API access with API key authentication

### PR DESCRIPTION
**Pull request by damilola-elegbede**
## Summary

Add REST API endpoints for external integrations (e.g., OpenClaw) with API key authentication, enabling programmatic access to all admin functionality via `/api/v1/*` endpoints.

### API Key Management
- Redis-based API key storage with secure SHA-256 hashing
- Admin UI for creating, enabling/disabling, and revoking keys
- Key format: `dk_live_*` (production) / `dk_test_*` (preview)

### V1 API Endpoints
- `GET /api/v1/stats` - Dashboard statistics
- `GET /api/v1/chats`, `/api/v1/chats/[id]` - Chat archive
- `GET /api/v1/fit-assessments`, `/api/v1/fit-assessments/[id]` - Fit assessments
- `POST /api/v1/fit-assessment` - Run fit assessment (JSON response)
- `GET /api/v1/resume-generations`, `/api/v1/resume-generations/[id]` - Resumes
- `GET /api/v1/audit` - Audit events with filtering
- `POST /api/v1/chat` - Chat with AI (JSON response)
- `GET /api/v1/traffic`, `/api/v1/usage` - Analytics (501 placeholder)

### Security & Audit
- Bearer token and X-API-Key header authentication
- SSRF protection for URL fetching (including redirect validation)
- API access logged with `accessType: "api"` distinction
- Consistent error response format with standard codes

### Accessibility Improvements
- aria-current for active navigation links
- aria-live regions for loading/error states
- Semantic table attributes and aria-labels

## Testing

15 new test files with comprehensive coverage:
- Library tests: api-response, api-key-storage, api-key-auth
- Admin API tests: api-keys CRUD operations
- V1 API tests: all endpoints with auth, filtering, pagination
- All 2085 tests passing

## Documentation

- `docs/external-api.md` - Complete API documentation with examples
- README updated with External API Access section

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Added External API with REST endpoints for accessing platform data (chats, assessments, resume generations, statistics, audit logs).
  * Added API Keys management portal in admin panel for generating and managing API keys.
  * Enhanced audit logs to track both browser and API-based access separately.

* **Documentation**
  * Added comprehensive External API documentation covering authentication, endpoints, rate limits, and security best practices.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by damilola-elegbede** [comment]
## Review Issue Acknowledgments

The following low-priority issues were reviewed locally and intentionally deferred:

### Low Priority (11 issues)

| Location | Issue | Reason |
|----------|-------|--------|
| `stats/route.ts:73` | Resume status only fetches first page | Acceptable for dashboard stats |
| `chat/route.ts:10` | Outdated Anthropic beta header | Non-breaking, address separately |
| `api-key-storage.ts:145` | listApiKeys loads all keys | Acceptable for <100 keys |
| `api-key-auth.ts:67` | lastUsedAt fire-and-forget | Non-critical tracking |
| `audit/route.ts:90` | In-memory sort for large sets | Pagination limits impact |
| `api-key-storage.ts:89` | Prefix extraction timing | Hash comparison is timing-safe |
| `admin/api-keys/route.ts:45` | No name uniqueness check | Names are informational only |
| `usage-logger.ts` | No retry mechanism | Non-critical data |
| `v1-chat.test.ts:101` | Test noted as bug | Bug was fixed in this PR |
| `chat/route.ts:55` | Content-Length NaN edge | Minimal impact edge case |
| `external-api.md:380` | Missing cursor expiration docs | Enhancement for later |

---
@coderabbitai These issues were reviewed during local development. No action needed.

File: docs/external-api.md [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+curl -H "Authorization: Bearer dk_live_xxx" \
+  https://damilola.tech/api/v1/stats
+```
+
+### X-API-Key Header
+
+```bash
+curl -H "X-API-Key: dk_live_xxx" \
+  https://damilola.tech/api/v1/stats
+```
```
[the remark this answers is not in the document]
Comment (damilola-elegbede) [reply docs/external-api.md]: @coderabbitai resolve - Replaced dk_live_xxx with YOUR_API_KEY across all examples

File: src/app/api/admin/api-keys/route.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    );
+  }
+
+  try {
+    const body = await req.json();
+    const { name, description } = body;
+
+    if (!name || typeof name !== 'string' || name.trim().length === 0) {
+      return Response.json({ error: 'Name is required' }, { status: 400 });
+    }
```
[the remark this answers is not in the document]
Comment (damilola-elegbede) [reply src/app/api/admin/api-keys/route.ts]: @coderabbitai resolve - Added try/catch for JSON parse errors in POST handler

File: src/app/api/v1/chats/[id]/route.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  const ip = getClientIp(req);
+
+  try {
+    const { id } = await params;
+    const blobUrl = decodeURIComponent(id);
+
+    // Validate URL to prevent SSRF attacks
+    if (!isValidBlobUrl(blobUrl)) {
+      return Errors.badRequest('Invalid chat URL');
+    }
```
[the remark this answers is not in the document]
Comment (damilola-elegbede) [reply src/app/api/v1/chats/[id]/route.ts]: @coderabbitai resolve - Added try/catch for decodeURIComponent URIError

File: src/app/api/v1/fit-assessment/route.ts [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    if (contentLength && parseInt(contentLength, 10) > MAX_BODY_SIZE) {
+      return Errors.badRequest('Request body too large.');
+    }
+
+    const body = await req.json();
+    const input = body.input || body.prompt;
+
+    if (!input || typeof input !== 'string') {
+      return Errors.validationError('Job description or URL is required in "input" field.');
+    }
```
[the remark this answers is not in the document]
Comment (damilola-elegbede) [reply src/app/api/v1/fit-assessment/route.ts]: @coderabbitai resolve - Added try/catch for JSON parse errors, returns 400

File: src/app/api/v1/traffic/route.ts [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+      }
+    } else {
+      const rawDays = Number.parseInt(searchParams.get('days') ?? '30', 10);
+      if (!Number.isFinite(rawDays)) {
+        return Errors.validationError('Invalid days parameter');
+      }
+      const days = Math.min(365, Math.max(1, rawDays));
+      endDate = new Date();
+      startDate = new Date();
+      startDate.setDate(startDate.getDate() - days);
```
[the remark this answers is not in the document]
Comment (damilola-elegbede) [reply src/app/api/v1/traffic/route.ts]: @coderabbitai resolve - Fixed off-by-one: startDate now uses days-1

**Comment by damilola-elegbede** [comment]
## Addressed CodeRabbit Feedback

**Resolved 10 review comments**

### Security
- Replaced `dk_live_xxx` placeholders with `YOUR_API_KEY` in documentation to avoid gitleaks false positives

### Error Handling
- Added JSON parse error handling in admin API keys routes (PATCH/POST) - returns 400 instead of 500
- Added `decodeURIComponent` URIError guards in `/v1/chats/[id]` and `/v1/resume-generations/[id]` - returns 400 for malformed URLs
- Added JSON body parse error handling in `/v1/fit-assessment` route - returns 400 for malformed JSON

### Bug Fix
- Fixed off-by-one in traffic route days calculation - `startDate.setDate(endDate.getDate() - (days - 1))`

### Code Quality
- Removed unused `data` variables in 3 test files to satisfy lint

---

### Skipped (1 issue)
| Location | Issue | Reason |
|----------|-------|--------|
| docs/external-api.md:92 | Markdownlint MD040/MD060 (table spacing) | Style preference, tables render correctly |

@coderabbitai These issues were reviewed during local development. No action needed.
